maas/resource_maas_vm_host_test.go

```
@@ -0,0 +1,122 @@
+package maas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"terraform-provider-maas/maas"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccMAASVMHost_DeployParams(t *testing.T) {
+	// A VM host identifier. Used to create a VM, which is deployed as a VM host in this test.
+	vmHostIdentifier := os.Getenv("TF_ACC_VM_HOST_MACHINE")
+	// A random string to be used for test
+	rs := acctest.RandString(8)
+
+	var defaultOS string
+	var defaultDistroSeries string
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
```

TF_ACC_VM_HOST_ID should also make the test fail if absent